2021-03-25 14:24:46.183328: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libnvinfer.so.6'; dlerror: libnvinfer.so.6: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /software/cuda/10.1/cudnn/7.6.4/lib64:/software/cuda/10.1/lib64:/software/python-3.6.2/gcc/lib
2021-03-25 14:24:46.183660: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libnvinfer_plugin.so.6'; dlerror: libnvinfer_plugin.so.6: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /software/cuda/10.1/cudnn/7.6.4/lib64:/software/cuda/10.1/lib64:/software/python-3.6.2/gcc/lib
2021-03-25 14:24:46.183696: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:30] Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
25-03-21 14:25 INFO: Loading faiss.
25-03-21 14:35 INFO: Loaded dataset of shape: (354902, 4099)
25-03-21 14:35 INFO: Dataset /storage/brno6/home/tslaninakova/learned-indexes/MTree1M-mocap loaded
2021-03-25 14:35:32 INFO: Benchmark initialized
25-03-21 14:35 INFO: Benchmark initialized
2021-03-25 14:35:32 INFO: Buckets: [5, 82]
25-03-21 14:35 INFO: Buckets: [5, 82]
2021-03-25 14:35:32 INFO: Total specs to train: 9
25-03-21 14:35 INFO: Total specs to train: 9
25-03-21 14:35 INFO: Training KMeans model with values of shape (354902, 4096): n. of clusters=5 | initializations=1 | max iterations=5
25-03-21 14:37 INFO: Training level 1
25-03-21 14:37 INFO: Training KMeans model with values of shape (63533, 4096): n. of clusters=82 | initializations=1 | max iterations=5
25-03-21 14:39 INFO: Training KMeans model with values of shape (59273, 4096): n. of clusters=82 | initializations=1 | max iterations=5
25-03-21 14:41 INFO: Training KMeans model with values of shape (47874, 4096): n. of clusters=82 | initializations=1 | max iterations=5
25-03-21 14:42 INFO: Training KMeans model with values of shape (98853, 4096): n. of clusters=82 | initializations=1 | max iterations=5
25-03-21 14:46 INFO: Training KMeans model with values of shape (85369, 4096): n. of clusters=82 | initializations=1 | max iterations=5
2021-03-25 14:49:03 INFO: KMeans with training spec: {'KMeans': [{'n_clusters': 5, 'n_init': 1, 'max_iter': 5}, {'n_clusters': 82, 'n_init': 1, 'max_iter': 5}]} trained on /storage/brno6/home/tslaninakova/learned-indexes/MTree1M-mocap in 810.27772388421 seconds (13.5046287314035 minutes)
25-03-21 14:49 INFO: KMeans with training spec: {'KMeans': [{'n_clusters': 5, 'n_init': 1, 'max_iter': 5}, {'n_clusters': 82, 'n_init': 1, 'max_iter': 5}]} trained on /storage/brno6/home/tslaninakova/learned-indexes/MTree1M-mocap in 810.27772388421 seconds (13.5046287314035 minutes)
Traceback (most recent call last):
  File "mocap_performance_kmeans_regular_no-AE.py", line 5, in <module>
    benchmark.evaluate()
  File "/auto/brno6/home/wollf/learned-indexes/learned-indexes/Benchmark.py", line 271, in evaluate
    recall = evaluate_knn_per_query(search_result, df_result, knns)
  File "/auto/brno6/home/wollf/learned-indexes/learned-indexes/knn_search.py", line 108, in evaluate_knn_per_query
    print(f"Evaluating k-NN performance on {len(res['objects_checkpoints'])} checkpoints: {res['objects_checkpoints']}")
KeyError: 'objects_checkpoints'
